examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -120,108 +152,75 @@ std::array<int, 6> RegularGridDecomposition::getExtentOfSubdomain(const int subd
 }
 
 void RegularGridDecomposition::exchangeHaloParticles(SharedAutoPasContainer &autoPasContainer) {
-  for (int i = 0; i < _dimensionCount; ++i) {
-    std::vector<ParticleType> haloParticles{};
-    for (int j = i; j < _dimensionCount; ++j) {
-      const size_t dimensionIndex = j % _dimensionCount;
-      std::vector<ParticleType> particlesForLeftNeighbour{};
-      std::vector<ParticleType> particlesForRightNeighbour{};
-
-      if (j == i) {
-        collectHaloParticlesForLeftNeighbour(autoPasContainer, dimensionIndex, particlesForLeftNeighbour);
-        collectHaloParticlesForRightNeighbour(autoPasContainer, dimensionIndex, particlesForRightNeighbour);
-      }
-
-      double leftHaloMin = _localBoxMin[dimensionIndex] - _skinWidth;
-      double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + _skinWidth;
-      double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - _skinWidth;
-      double rightHaloMax = _localBoxMax[dimensionIndex] + _skinWidth;
-
-      for (const auto &particle : haloParticles) {
-        std::array<double, _dimensionCount> position = particle.getR();
-        if (position[dimensionIndex] >= leftHaloMin and position[dimensionIndex] < leftHaloMax) {
-          particlesForLeftNeighbour.push_back(particle);
-
-          // Apply boundary condition
-          if (_localBoxMin[dimensionIndex] == _globalBoxMin[dimensionIndex]) {
-            position[dimensionIndex] =
-                position[dimensionIndex] + (_globalBoxMax[dimensionIndex] - _globalBoxMin[dimensionIndex]);
-            particlesForLeftNeighbour.back().setR(position);
-          }
-        } else if (position[dimensionIndex] >= rightHaloMin and position[dimensionIndex] < rightHaloMax) {
-          particlesForRightNeighbour.push_back(particle);
-
-          // Apply boundary condition
-          if (_localBoxMax[dimensionIndex] == _globalBoxMax[dimensionIndex]) {
-            position[dimensionIndex] =
-                position[dimensionIndex] - (_globalBoxMax[dimensionIndex] - _globalBoxMin[dimensionIndex]);
-            particlesForRightNeighbour.back().setR(position);
-          }
+  std::vector<ParticleType> haloParticles{};
+
+  for (int dimensionIndex = 0; dimensionIndex < _dimensionCount; ++dimensionIndex) {
+    std::vector<ParticleType> particlesForLeftNeighbor{};
+    std::vector<ParticleType> particlesForRightNeighbor{};
+
+    collectHaloParticlesForLeftNeighbor(autoPasContainer, dimensionIndex, particlesForLeftNeighbor);
+    collectHaloParticlesForRightNeighbor(autoPasContainer, dimensionIndex, particlesForRightNeighbor);
```

Comment:
shouldn't `particlesForLeftNeighbor` and `particlesForRightNeighbor` be somehow merged into `haloParticles`? Otherwise the latter is empty and we never enter the upcoming loop.